### PR TITLE
Scripts/Outland: migrate Elder Kuruti to SAI

### PR DESCRIPTION
**Pull request by mysterious-static**
**Changes proposed:**

- Migrates NPC Elder Kuruti to SmartAI from creature script.

**Target branch(es):** 3.3.5/master

- [X] 3.3.5
- [X] master

**Tests performed:** (Does it build, tested in-game, etc.)

Builds. Tested ingame, NPC behaves as you would expect. Has gossip and gives you the item if you have the quest and don't already have the item, doesn't otherwise.

**Known issues and TODO list:** (add/remove lines as needed)

- [x] Rename SQL to 9999_99_99, I'll do that in a second

### CODE REVIEW

**Comment by tkrokli** [comment]
Suggestion: remove the SQL comments as long as they don't add any useful information to the file, save for a blank comment ( `--` ) on top of the file.

**Comment by velinath** [comment]
sorry, no idea what happened on that rebase

File: sql/updates/world/3.3.5/2017_09_25_99_world.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
+--
+DELETE FROM `gossip_menu` WHERE `MenuID` BETWEEN 57020 AND 57022;
+INSERT INTO `gossip_menu` (`MenuID`, `TextID`, `VerifiedBuild`) VALUES
+(57020,9227,0),(57021,9229,0),(57022,9231,0);
```
Comment (Killyana) [inline sql/updates/world/3.3.5/2017_09_25_99_world.sql]: This gossips are already on the DB, just use them, no need to create new ones:
 7583, 9227
 7585, 9229
 7586, 9231

Comment (tkrokli) [reply sql/updates/world/3.3.5/2017_09_25_99_world.sql]: Oh. Right. I forgot to look up those.
```sql
SELECT * FROM `gossip_menu` WHERE `TextID` IN (9227,9229,9231);
```
```
MenuID  TextID  VerifiedBuild
------  ------  -------------
  7583    9227              0
  7585    9229              0
  7586    9231              0
```

Comment (velinath) [reply sql/updates/world/3.3.5/2017_09_25_99_world.sql]: rusty I guess, my bad!

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+--
+DELETE FROM `gossip_menu` WHERE `MenuID` IN (7583,7585,7586);
+INSERT INTO `gossip_menu` (`MenuID`, `TextID`, `VerifiedBuild`) VALUES
+(7583,9227,0),(7585,9229,0),(7586,9231,0);
```
Comment (tkrokli) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql]: I think you misunderstood it.
Remove this section completely.

Comment (velinath) [reply sql/updates/world/3.3.5/9999_99_99_99_world.sql]: ah yeah that makes sense

File: sql/updates/world/3.3.5/9999_99_99_99_world.sql [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+(7582,0,0,'Greetings, elder. It is time for your people to end their hostility toward the draenei and their allies.',14872,1,3,7583,0,0,0,'',0,-1),
+(7583,0,0,'I did not mean to deceive you, elder. The draenei of Telredor thought to approach you in a way that would be familiar to you.',14874,1,3,7585,0,0,0,'',0,-1),
+(7585,0,0,'I will tell them. Farewell, elder.',14879,1,3,7586,0,0,0,'',0,-1);
+
+UPDATE `creature_template` SET `AIName` = 'SmartAI', `ScriptName` = '' WHERE `entry` = 18197;
+
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 18197);
+INSERT INTO `smart_scripts` (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
+(18197, 0, 0, 0, 62, 0, 100, 0, 7585, 0, 0, 0, 56, 24573, 1, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Elder Kuruti - On Gossip Option 0 Selected - Add Item \'Elder Kuruti\'s Response\'  1 Time');
```
Comment (Killyana) [inline sql/updates/world/3.3.5/9999_99_99_99_world.sql]: Use this spell to add the item: https://woehead.way-of-elendil.fr/?spell=32152

Comment (velinath) [reply sql/updates/world/3.3.5/9999_99_99_99_world.sql]: thanks for catching. i think this got screwed up when i synced the pr a couple of days ago. apologies

iirc it worked without setting castflags during testing :)

**Comment by tkrokli** [comment]
Just a little extra tip, in case you find it useful: if you use double quotes ` " " ` around text lines / comment lines, you can skip the escape backslash ( ` \' ` ) for single quotes or apostrophes inside the text comments.

**Review by Killyana — APPROVED** [review]

**Comment by velinath** [comment]
for sure, i personally prefer single quotes - if TC folks prefer double quotes I can change going forward?

**Comment by tkrokli** [comment]
Well, choose what you prefer to use for yourself, it is not a big deal. Myself, I find it useful to put double quotes around spoken sentences and SAI comments (to avoid using escape backslash) _only_ when I find existing apostrophes/ single quotes as part of the text, either as contractions in speech or in possessive nouns (external references) and use single quotes when there is no need to use escape (backslash). This is mostly about keeping text readability (cosmetic), so for all I know, you are free to choose which format to use. Consistency is usually best.

**Review by ghost — APPROVED** [review]

**Comment by velinath** [comment]
i believe these were changes that disappeared when i rebased, but it should be good now

i'm going to refrain from touching the rebase button further

**Comment by tkrokli** [comment]
Thank you for the update. PR content rebase is usually not needed in a PR since it will be squashed upon merge anyway, so it is normally more useful to keep as many commits as you can in the PR to be able to look up the changes before a merge is done at the end. I presume a rebase will only be useful in the case of an external update to the included source files in the PR, but I suppose it is possible to add the changes manually too in such a case. In this PR, it is less likely to happen, unless someone performs a general script content change for many scripts at the same time.

**Pull request merged**